src/GitHub.App/Factories/ViewViewModelFactory.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Windows;
+using GitHub.Exports;
+using GitHub.Services;
+using GitHub.ViewModels;
+
+namespace GitHub.Factories
+{
+    /// <summary>
+    /// Factory for creating views and view models.
+    /// </summary>
+    [Export(typeof(IViewViewModelFactory))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class ViewViewModelFactory : IViewViewModelFactory
+    {
+        readonly IGitHubServiceProvider serviceProvider;
+
+        [ImportingConstructor]
+        public ViewViewModelFactory(
+            IGitHubServiceProvider serviceProvider,
+            ICompositionService cc)
+        {
+            this.serviceProvider = serviceProvider;
+            cc.SatisfyImportsOnce(this);
```

Comment:
I don't think importing `ICompositionService` or calling `SatisfyImportsOnce` is necessary. This should happen anyway after `ViewViewModelFactory` has been created.

I'm not sure `AllowRecomposition = true` is supported in Visual Studio's MEF implementation. I'm guessing this class some from somewhere else?

Reply:
Not sure - this class used to be [ExportFactoryProvider](https://github.com/github/VisualStudio/blob/e13678c2ea2bbc1cbefa3ed171b6bbf025569040/src/GitHub.Exports/Services/ExportFactoryProvider.cs#L15) and the code comes from there.

Reply:
I put a breakpoint on the `set` method of `Views` and this property is indeed being set twice (with the same number of elements).

Reply:
@sguthals Done, see #1368.